mesonbuild/mtest.py

```diff
@@ -520,17 +519,20 @@ def __init__(self) -> None:
             self.sub = '| '
             self.spinner = self.ASCII_SPINNER
 
-    def flush(self) -> None:
-        if self.should_erase_line:
-            print(self.should_erase_line, end='')
-            self.should_erase_line = ''
+    @classmethod
```

Comment:
This is a bad idea. The ConsoleLogger has a single instance, so there's no need to move things to the class side. If you need to access it from elsewhere keep the methods in the instance and add a `get_instance()` class method to retrieve the singleton.

Reply:
You are right. That is a better approach.

Reply:
Hopefully the end result is even better with no class side functions or even `get_instance()`.